File: src/Api/Model/DB/Endereco.cs
using Newtonsoft.Json;
using Realms;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Text;

namespace CriadorBaseDados.Model.DB
{
    [JsonObject(MemberSerialization.OptIn)]
    public class Endereco : RealmObject
    {
        [JsonProperty]
        public string Logradouro { get; set; }
        [JsonProperty]
        public string Numero { get; set; }
        [DefaultValue("")]
        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
        public string Complemento { get; set; }
        [DefaultValue("")]
        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
        public string Bairro { get; set; }
        [JsonProperty]
        public string CEP { get; set; }
    }
}

File: src/Mobile/b2b/lib/model/core/endereco.dart
class Endereco {
  final String id;
  final String logradouro;
  final String numero;
  final String complemento;
  final String bairro;
  final String cep;

  Endereco(this.id, this.logradouro, this.numero, this.complemento, this.bairro, this.cep);

  Endereco.fromJson(this.id, Map<String, dynamic> json)
      : logradouro = json['logradouro'],
        numero = json['numero'],
        complemento = json['complemento'] ?? "",
        bairro = json['bairro'],
        cep = json['cep'];

  Map<String, dynamic> toMap() => <String, dynamic>{
        'id': id,
        'logradouro': logradouro,
        'numero': numero,
        'complemento': complemento,
        'bairro': bairro,
        'cep': cep,
      };
}
